tests/core/test_u_cache.py

```diff
@@ -0,0 +1,40 @@
+from pytest import fixture
+from numpy import asarray as array
+from numpy.testing import assert_equal
+
+from openfisca_core.cache import Cache
+from openfisca_core import periods
+
+period = periods.period(2018)
+
+
+@fixture
+def cache():
```

Comment:
> Some methodology question for @Morendil :
> 
> Let's assume that `Cache` uses, in its implementation, a bunch of `InMemoryStorage` instances to store data. This is `Cache` internal business, and is not reflected in the public methods.
> 
> To write a proper **unit** test on Cache, do I need to stub/mock `InMemoryStorage`, or can I just write tests on the Cache methods, as below?
> 
> In other words, can a unit test indirectly run some code that is outside of the class under test, or does it make it an integration test ?
> 
> Thanks a lot in advance!

Reply:
I try not to get too hung up on unit/integration test differences, it's a tactical matter rather than one of principle. It's fine if the methods of the object under test rely on some other collaborators, as long as you're primarily testing that object's behaviour, rather than indirectly testing the collaborators.

Reply:
Ok, thanks a lot for the explanation 🙂